content/chat/participant-activity.textile

```diff
@@ -1,7 +1,220 @@
+
+```[realtime_javascript]
+channel.presence.subscribe(‘leave’, (member) => {
+  user.online = false
+  user.lastOnline = member.timestamp
+
+  // Custom functionality to update display for chat participant that another participant is no longer online
+});
+```
+
+h3(#update). Update
+
+In certain chat instances, chat members can associate their chat participant status with a text-based status. For example, if you have a chat participant object (user) with the @statusMessage@ field, you can trigger an @update@ event using the code sample below to update this user with a new text-based string.
+
+```[realtime_javascript]
+channel.presence.update('<NEW_STATUS_UPDATE>', (err) => {
+  user.statusMessage = '<NEW_STATUS_UPDATE>';
+});
+```
+
+When a chat participant triggers the @update@ event all subscribers for this presence channel will be notified in real-time allowing them to update the string value of that chat participant's status of this user for everyone.
+
+```[realtime_javascript]
+channel.presence.subscribe('update', (memberUpdate) => {
+  user.status = memberUpdate.data
+});
+```
+
+h3(#other-events). Other events / more information
+
+The @present@ event is not relevant for chat participant statuses, however, if you wish to read more about other functionalities of presence, please refer to the "presence":/presence-occupancy/presence?lang=javascript#trigger-events docs page.
+
+h2(#subscribe). Subscribe chat participants to events
+
+Chat participant clients can subscribe to the presence set, enabling them to receive updates on other participants and update the UI accordingly (e.g., displaying online status icons).
+
+The message count within a presence set, like online status, can surge as the participant number grows. For instance, with 200 participants rapidly joining and leaving, the potential messages for online status alone could reach 80,400. The breakdown includes:
+
+200 chat participant messages published for the @enter@ event.
+200 × 200 (40,000) messages subscribed to for the @enter@ events.
+200 chat participant messages published for the @leave@ event.
+200 × 200 (40,000) presence messages subscribed to for the @leave@ event.
+
+For improved scalability, consider making your backend server the exclusive subscriber to chat participants' online status updates if your chat application anticipates a large number of participants. In this scenario, frontend clients can periodically request updates from the backend.
+
+h3(#batch-presence). Batch presence retrieval
+
+It is possible to reduce realtime presence updates received and traffic between clients by introducing batches using the @request()@ in the REST API. Within a batch response is an array of presence states, where each presence state update contains information about the clients in the presence set, including their @clientId@, member data, and "presence action":/docs/api/realtime-sdk/types#presence-action .
```

Comment:
Perhaps provide an example i.e. "for certain use cases like large-scale, live streaming chat it may not be desirable or efficient to send individual presence updates..."

Reply:
Updated: https://github.com/ably/docs/pull/2090/files#diff-7e6b73a6da0656177108f5d85ad297b6370a62c6419a06e443a906e81765658eR99